code/modules/reagents/reagent_containers/food/snacks/grown.dm

```diff
@@ -41,6 +43,206 @@
+		if(affecting)
+			if(thorns_apply_damage(H, affecting))
+				H << "<span class='danger'>Your mouth is cut by \the [src]'s sharp thorns!</span>"
+				//H.stunned++ //just a 1 second pause to prevent people from spamming pagedown on this, since it's important
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/examine(mob/user)
+	..()
+	if(!seed) return
+	var/traits = ""
+	if(seed.stinging) traits += "<span class='alert'>It's covered in tiny stingers.</span> "
+	if(seed.thorny) traits += "<span class='alert'>It's covered in sharp thorns.</span> "
+	if(seed.juicy == 2) traits += "It looks ripe and excessively juicy. "
+	if(seed.teleporting) traits += "It seems to be spatially unstable. "
+	if(traits) user << traits
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/splat_decal(turf/T)
+	var/obj/effect/decal/cleanable/S = getFromPool(seed.splat_type,T)
+	S.New(S.loc)
+	if(seed.splat_type == /obj/effect/decal/cleanable/fruit_smudge/)
+		if(filling_color != "#FFFFFF")
+			S.color = filling_color
+		else
+			S.color = AverageColor(getFlatIcon(src, src.dir, 0), 1, 1)
+		S.name = "[seed.name] smudge"
+	if(seed.biolum && seed.biolum_colour)
+		S.set_light(1, l_color = seed.biolum_colour)
+	return 1
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/splat_reagent_reaction(turf/T)
+	if(src.reagents.total_volume > 0)
+		src.reagents.reaction(T)
+		for(var/atom/A in T)
+			src.reagents.reaction(A)
+		return 1
+	return 0
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/thorns_apply_damage(mob/living/carbon/human/H, datum/organ/external/affecting)
+	if(!seed.thorny || !affecting)
+		return 0
+	//if(affecting.take_damage(5+seed.carnivorous*3, 0, 0, "plant thorns")) //For some fucked up reason, it's not returning 1
+	affecting.take_damage(5+seed.carnivorous*3, 0, 0, "plant thorns")
+	H.UpdateDamageIcon()
+	return 1
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/stinging_apply_reagents(mob/living/carbon/human/H)
+	if(!seed.stinging)
+		return 0
+	if(!reagents || reagents.total_volume <= 0)
+		return 0
+	if(!seed.chems || !seed.chems.len)
+		return 0
+	var/injecting = Clamp(1, 3, potency/10)
+	for(var/rid in seed.chems) //Only transfer reagents that the plant naturally produces, no injecting chloral into your nettles.
+		reagents.trans_id_to(H,rid,injecting)
+		. = 1
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/do_fruit_teleport(atom/hit_atom, mob/M, var/potency)
+	//Does this need logging?
+	var/outer_teleport_radius = potency/10 //Plant potency determines radius of teleport.
+	var/inner_teleport_radius = potency/15 //At base potency, nothing will happen, since the radius is 0.
+	var/list/turfs = new/list()
+	var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+	if(inner_teleport_radius < 1)
+		return 0
+	//This could likely use some standardization but I have no idea how to not break it.
+	for(var/turf/T in orange(M,outer_teleport_radius))

```

Comment:
This is gonna need some sanity for Centcomm (look at how bluespace crystals do it)